security/pkg/nodeagent/sds/server.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sds
+
+import (
+	"net"
+
+	"google.golang.org/grpc"
+
+	"istio.io/istio/pkg/log"
+)
+
+const maxStreams = 100000
+
+// Args provides all of the configuration parameters for secret discovery service.
+type Args struct {
+	SDSUdsSocket string
+}
+
+// Server is the grpc server that exposes SDS through UDS.
+type Server struct {
+	envoySds          *sdsservice
+	grpcServer        *grpc.Server
+	grpcListeningAddr net.Addr
+
+	closing chan bool
+}
+
+// NewServer creates and starts the Grpc server for SDS.
+func NewServer(args Args, st secretStore) (*Server, error) {
```

Comment:
We should make secretStore public, since it's a part of the public method `NewServer` signature.

Also question, do you plan to make the SecretStore is shared between this module and the the module where node agent talks to CA to get the key cert?

Reply:
yes, it should be public, it's OK to put it private for this PR since it's a placeholder.

Moving forward my draft plan is, SecretStore will be in its own pkg, it manages key/cert in memory, has a separated refresher job for key rotation, refers a caClient interface for CSR(caClient interface and its two implementations(oss, csm) in separated pkg).

Reply:
sounds good, we can change it to public by then.